src/js/main.js

```diff
@@ -0,0 +1,201 @@
+              chooseChildAgeOption.innerText = 'Choose child age';
+
+              select.append(chooseChildAgeOption);
+
+              for (let i = 0; i < 18; i++) {
+                let option = document.createElement('option');
+                option.classList.add('child-select');
+                option.value = i;
+                option.innerText = `${i} years`;
+                option.setAttribute('data-index', i);
+
+                if (selected.children[childIndex] == i) {
+                  option.setAttribute('selected', true);
+                }
+
+                select.appendChild(option);
+              }
+              select.addEventListener('change', (e) => {
+                let target = e.target;
+                selected.children[target.getAttribute('data-index')] = parseInt(
+                  target.value,
+                );
+              });
+              childrenBlock.appendChild(select);
+              div.append(childrenBlock);
+            }
+          }
+        },
+      },
+    },
+  },
+  container = document.getElementById('modal-filter');
+
+function render(container, elements) {
+  function buildBlock(element) {
+    let div = document.createElement('div'),
+      divContent = document.createElement('div'),
+      label = document.createElement('label'),
+      input = document.createElement('input'),
+      btnDecrease = document.createElement('button'),
+      btnIncrease = document.createElement('button'),
+      updateValue = function (div) {
+        if (element.hasOwnProperty('events')) {
+          let events = element.events;
+
+          if (events.hasOwnProperty('onValueUpdate')) {
+            events.onValueUpdate(element, container);
+          }
+        }
+      };
+
+    div.setAttribute('data-div', `${element.identifier}-block`);
+    div.classList.add('check--form_content');
+    divContent.classList.add('check-stepper');
+    label.innerText = element.label;
+    label.classList.add('check-name');
+    input.classList.add('check-count');
+    btnDecrease.innerText = '-';
+    btnIncrease.innerText = '+';
+
+    input.value = element.value;
+
+    btnDecrease.setAttribute('data-update', element.identifier);
+    btnDecrease.setAttribute('type', 'button');
+    btnDecrease.className = 'btn-check-l';
+    btnDecrease.classList.add('active');
+
+    btnIncrease.setAttribute('data-update', element.identifier);
+    btnIncrease.setAttribute('type', 'button');
+    btnIncrease.className = 'btn-check-r';
+    btnIncrease.classList.add('active');
+
+    btnDecrease.addEventListener('click', (e) => {
+      if (element.value <= element.values.min) return;
+
+      elements[e.target.getAttribute('data-update')].value--;
+
+      let counter = document.querySelector(`[data-value='${element.identifier}']`);
+      counter.innerText = [element.value, element.label].join(' ');
+
+      container.innerHtml = '';
+      render(container, elements);
```

Comment:
Why do you re-render the whole form every time you change it? If you change the value, re-rendering is only required for one number inside the form block and the list of ages of children. You can try to use [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) for this